src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.PackGroup.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using static Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver;
+
+namespace Microsoft.DotNet.Workloads.Workload.Install
+{
+    internal partial class NetSdkMsiInstallerClient
+    {
+        class WorkloadPackGroupJson
+        {
+            public string GroupPackageId { get; set; }
+            public string GroupPackageVersion { get; set; }
+
+            public List<WorkloadPackJson> Packs { get; set; } = new List<WorkloadPackJson>();
+        }
+
+        class WorkloadPackJson
+        {
+            public string PackId { get; set; }
+
+            public string PackVersion { get; set; }
+        }
+
+        Dictionary<(string packId, string packVersion), List<WorkloadPackGroupJson>> GetWorkloadPackGroups()
+        {
+            Dictionary<(string packId, string packVersion), List<WorkloadPackGroupJson>> ret = new();
+
+            var manifests = _workloadResolver.GetInstalledManifests();
+            foreach (var manifest in manifests)
+            {
+                var packGroupFile = Path.Combine(manifest.ManifestDirectory, "WorkloadPackGroups.json");
```

Comment:
This is the new mapping file we're going to add to the manifest installers?

Reply:
Correct.